### PR TITLE
Add Signing Component

### PR DESCRIPTION
**Pull request by TommyHuang-dev**
This it the initial merge of signing component into master. The component will let users sign and submit a message into their virtrolio.

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -2,11 +2,13 @@ import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
 import { HomeComponent } from './home/home.component';
 import { PageNotFoundComponent } from './page-not-found/page-not-found.component';
+import { SigningComponent } from './signing/signing.component';
 
 const routes: Routes = [
   { path: '', component: HomeComponent },
-  { path: 'page-not-found', component: PageNotFoundComponent },
-  { path: '**', pathMatch: 'full', redirectTo: '/page-not-found'}
+  { path: 'pagenotfound', component: PageNotFoundComponent },
```
Comment (ricetech) [inline src/app/app-routing.module.ts:9]: You're using the incorrect routerLink for PNF. 
```suggestion
  { path: 'page-not-found', component: PageNotFoundComponent },
```

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -2,11 +2,13 @@ import { NgModule } from '@angular/core';
...
+  { path: 'signing', component: SigningComponent },
+  { path: '**', pathMatch: 'full', redirectTo: '/pagenotfound'}
```
Comment (ricetech) [inline src/app/app-routing.module.ts:11]: You're using the incorrect routerLink for PNF. 
```suggestion
  { path: '**', pathMatch: 'full', redirectTo: '/page-not-found'}
```

File: src/app/signing/signing.component.css [hunk 3]
```diff
@@ -0,0 +1,32 @@
+
+#signing-box {
+	width: 50em;
+	height: 10em;
+	resize: none; /*both;*/
```
Comment (ricetech) [inline src/app/signing/signing.component.css:5]: Remove unused code
```suggestion
	resize: none;
```

File: src/app/signing/signing.component.css [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	margin-top: 2em;
+	margin-left:auto;
+	margin-right:auto;
+	padding: 0.1em;
+}
+
+#preview-box {
+	width: 50em;
+	height: 10em;
+	resize: none; /*both;*/
```
Comment (ricetech) [inline src/app/signing/signing.component.css:15]: Remove unused code
```suggestion
	resize: none;
```

File: src/app/signing/signing.component.html [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  <div class="row">
+    <h3>Preview:</h3>
+  </div>
+  <div class="row">
+    <textarea class="signing-box" #previewbox id="preview-box" readonly></textarea>
+  </div>
+</div>
+
+
+<div class="modal fade" id="exampleModalCenter" tabindex="-1" role="dialog" aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
```
Comment (ricetech) [inline src/app/signing/signing.component.html:60]: Don't use default names.
```suggestion
<div class="modal fade" id="submitConfirmModal" tabindex="-1" role="dialog" aria-labelledby="submitConfirmModalTitle" aria-hidden="true">
```

File: src/app/signing/signing.component.html [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    <div class="modal-dialog modal-dialog-centered" role="document">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h5 class="modal-title" id="exampleModalLongTitle">Are you sure you want to submit?</h5>
```
Comment (ricetech) [inline src/app/signing/signing.component.html:64]: Don't use default names.
```suggestion
          <h5 class="modal-title" id="submitConfirmModalTitle">Are you sure you want to submit?</h5>
```

File: src/app/signing/signing.component.html [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  </div>
+  <div class="row">
+    <h3>Preview:</h3>
+  </div>
+  <textarea 
+  class="shadow"
+  id="signing-box" 
+  autofocus
+  ></textarea>
+  <div class="row">
```
Comment (GuyInFridge) [inline src/app/signing/signing.component.html:52]: Did you get around to fixing the test display inside of this div so it has a fixed width?

- Look in the Markdown library for solutions
- Use some css to manually limit the sizes of the paragraph tags inside the div

File: src/assets/css/main.css [hunk 8]
```diff
@@ -36,7 +36,6 @@ p {
   /* Set all components' [outermost] containers (or fluid-containers) to an appropriate minimum height which will fill the screen and push the footer down. vh is viewport height */
 }
 
-
```
Comment (GuyInFridge) [inline src/assets/css/main.css:39]: Remove this so we don't get merge conflicts later?

File: src/styles.css [hunk 9]
```diff
@@ -1 +1,3 @@
 /* You can add global styles to this file, and also import other style files */
+
```
Comment (GuyInFridge) [inline src/styles.css:2]: Remove this so we don't get merge conflicts later?

File: src/app/signing/signing.component.html (hunk 7 above)
Comment (TommyHuang-dev) [reply src/app/signing/signing.component.html:52]: Not yet, that will be fixed on subsequent merges

**Pull request merged**